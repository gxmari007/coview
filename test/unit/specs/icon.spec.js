import { shallow } from 'vue-test-utils';
import { createRenderer } from 'vue-server-renderer';
import CoIcon from '@/components/icon';

describe('CoIcon', () => {
  it('type prop', () => {
    const wrapper = shallow(CoIcon, {
      propsData: { type: 'ant-design' },
    });

    expect(wrapper.hasClass('anticon')).toBe(true);
    expect(wrapper.hasClass('anticon-ant-design')).toBe(true);
  });

  it('spin prop', () => {
    const wrapper = shallow(CoIcon, {
      propsData: { spin: true },
    });

    expect(wrapper.hasClass('anticon-spin')).toBe(true);
  });

  it('match snapshot', () => {
    const renderer = createRenderer();
    const wrapper = shallow(CoIcon, {
      propsData: { type: 'ant-design' },
    });

    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err);
      expect(str).toMatchSnapshot();
    });
  });
});
