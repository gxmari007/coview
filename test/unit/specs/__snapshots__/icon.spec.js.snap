// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoIcon match snapshot 1`] = `<i class="anticon anticon-ant-design"></i>`;
